pkg/types/azure/validation/platform.go

```diff
@@ -52,7 +52,7 @@ func ValidatePlatform(p *azure.Platform, publish types.PublishingStrategy, fldPa
 	if p.Region == "" {
 		allErrs = append(allErrs, field.Required(fldPath.Child("region"), "region should be set to one of the supported Azure regions"))
 	}
-	if !p.IsARO() && publish == types.ExternalPublishingStrategy {
+	if !p.IsARO() && publish != types.InternalPublishingStrategy {
```

Comment:
Nit: actually, publish: Mixed + apiserver: Internal + ingress: Internal is the similar scenario as private cluster (publish: Internal), it also doesn't require BaseDomainResourceGroupName. Do we need to consider it?